test/pkg.jl

```diff
@@ -743,4 +743,14 @@ end
     end end
 end
 
+# PR #1784 - Remove trailing slash from URL.
+@testset "URL with trailing slash" begin
+    temp_pkg_dir() do project_path
+        with_temp_env() do
+            Pkg.add(Pkg.PackageSpec(url = "https://github.com/JuliaLang/Example.jl.git/"))
```

Comment:
This doesn't test the clone path though? But maybe it failed the download path too?

Reply:
Without the `rstrip` patch it stops with a username prompt at least.

Reply:
And as far as I can tell, `add` from URL clones.

Reply:
Yea you are right, since we need to checkout master.